### ISSUE TITLE
hd_root stalling in ccdb load?

### ISSUE THREAD

**Issue by mikestaib** [JeffersonLab/sim-recon#132]
Latest version of the code stalls after "DParticleID: loading values from PID data base." Launched ~150 jobs all timeout. Seems like this is also happening to the offline monitoring run as most of the jobs have timed out. Using latest CCDB, JANA and sim-recon.

ifarm1401> hd_root -PPLUGINS=HLDetectorTiming /cache/mss/halld/RunPeriod-2015-03/rawdata/Run003185/hd_rawdata_003185_000.evio
JANA >>OUTPUT_FILENAME: hd_root.root
JANA >>Initializing plugin "/w/halld-scifs1a/home/mstaib/builds/sim-recon/Linux_CentOS6-x86_64-gcc4.4.7/plugins/HLDetectorTiming.so" ...
Opened ROOT file "hd_root.root" ...
JANA >>Opening source "/cache/mss/halld/RunPeriod-2015-03/rawdata/Run003185/hd_rawdata_003185_000.evio" of type: EVIO  - Reads EVIO formatted data from file or ET system
JANA >>Launching threads .

JANA >>Created JCalibration object of type: JCalibrationCCDB
JANA >>Generated via: JCalibration using CCDB for MySQL and SQLite databases
JANA >>Run:3185
JANA >>URL: sqlite:////work/halld2/home/mstaib/ccdb.sqlite
JANA >>context: default
JANA >>Reading translation table from calib DB: Translation/DAQ2detector ...
JANA >>23160 channels defined in translation table
Created TGeoManager :0x7f645883ea20
JANA >>Creating DGeometry:1.0 events read)  2.0Hz  (avg.: 0.0Hz)
JANA >>  Run requested:3185  found:3185
JANA >>  Run validity range: 3185-3185
JANA >>  URL="xmlfile:///work/halld2/home/mstaib/builds/hdds//main_HDDS.xml"  context="default"
JANA >>  Type="JGeometryXML"
JANA >>Found 18 material maps in calib. DB for run 3185
JANA >>Read in 18 material maps containing 62173 grid points total
DParticleID: loading values from PID data base

**Comment by pmattjlab** [comment]
Offline monitoring just switched to JANA 0.7.4 and CCDB 1.06.00.  Is that what you're using too?

**Comment by mikestaib** [comment]
I am using the head of JANA and CCDB, so I'd imagine it's close enough...can grab the tagged versions to check. Looks like the offline monitoring jobs aren't doing so hot though?

**Comment by pmattjlab** [comment]
Monitoring jobs are timing out, but it looks like they aren't even really starting, so this may be separate.

**Comment by sdobbs** [comment]
Try using the MySQL master too - there could be a disk issue. I'll try
running some checks when I get back to my laptop...

**Comment by faustus123** [comment]
Hi Guys,

  I think this may be a bug in JANA 0.7.4. I just did a quick test and it looked to hang.
There was a redirect left over from the 64 bit event number change that should have been removed
but wasn’t. When I removed it, things seemed to run OK. I’ll look into it a bit more and
then put a patched version of JANA 0.7.4 up.

Regards,
-David

**Comment by sparksn** [comment]
Running hd_root (a0c71ac) hangs on me too when using CCDB-1.05.00 and JANA 0.7.4. When I switch back to JANA 0.7.3, the same code crashes where it previously hanged with 
`terminate called after throwing an instance of ‘unsigned long'`
`Abort`
Running hd_root with the same plugins attached and compiled against the same dependencies works fine using sim-recon 1090bb5.
This indicates that there were bugs introduced by sim-recon commit a0c71ac.

**Comment by faustus123** [comment]
Hi All,

  For those not following this issue thread on github: there have been hangs of sim-recon
jobs reported. This was after the recent JANA 0.7.4 upgrade and the corresponding
sim-recon change to support 64bit event numbers.

  A patched version tagged JANA 0.7.4p1 has been posted to address the issue.
I have built this on ifarm1401 for the Linux_CentOS6-x86_64-gcc4.4.7
and Linux_CentOS6-x86_64-clang3.7.0 compilers. It can be found in:

/group/12gev_phys/builds

Note that this version was built against the newest tagged version of CCDB,
v1.06.00.

I did not build it in /group/halld/Software/builds since it appears there is some
massaging of the directory structure there from the default. I’ll let others take
care of that.

  The 0.7.4p1 version also addresses an issue with compiling on the latest
version of OS X “El Capitan”. (Thanks to Nathan for researching the issue
and providing a solution.)

Anyone who has recently upgraded to 0.7.4 will need to upgrade again
to 0.7.4p1 (sorry).

Regards,
-David
